src/plugins/data/public/antlr/shared/utils.ts

```diff
@@ -93,24 +96,27 @@ export const fetchData = (
   });
 };
 
-// Specific fetch function for table schemas
-// TODO: remove this after using data set table schema fetcher
-export const fetchTableSchemas = (tables: string[], api: any, queryString: QueryStringContract) => {
-  return fetchData(
-    tables,
-    (table, dataSourceId, title) => ({
-      query: { query: `DESCRIBE TABLES LIKE ${table}`, format: 'jdbc' },
-      df: {
-        meta: {
-          queryConfig: {
-            dataSourceId: dataSourceId || undefined,
-            title: title || undefined,
-          },
-        },
+export const fetchColumnValues = (
+  tables: string[],
+  column: string,
+  services: IDataPluginServices,
+  dataset?: Dataset
+) => {
+  if (!services.uiSettings.get(UI_SETTINGS.QUERY_ENHANCEMENTS_SUGGEST_VALUES)) {
+    return new Promise(() => []);
```

Comment:
should this guard happen before calling `fetchColumnValues`? so this function is not coupled with the setting

Reply:
I think it might be better kept this way since fetchColumnValues is fully expected by its callers to return nothing if there's any reason why columns should not be returned, for example if the field type isn't supported.